### PR TITLE
Add Docker, SAML docs

### LINKED ISSUE #746: Document device private key generation
Hey,

would it be possible to allow an expert setting that experienced users can provide there own public key generated offline, so the security level is higher?

Best regards
Mike

### LINKED ISSUE #794: Better document how to use external DBs
Document using `ssl_opts` and different configurations, giving an example configuration.

### LINKED ISSUE #913: Docker deployment docs
Issue to track adding docs around our new Docker image in production.

Current docs need to be made context-aware for situations where Docker and Omnibus differ. E.g. `firezone-ctl` is no longer the way to reload configuration.

refs #260 

@princemaple 's instructions for setting up a working Firezone instance in Docker: Refs #887

### LINKED ISSUE #959: Document how to regenerate portal secret keys
Document how to regenerate:

- [ ] DB encryption key
- [ ] phoenix salts / secrets
- [ ] WireGuard private key

### LINKED ISSUE #960: SAML docs
refs #926

### PR DESCRIPTION
**Pull request by jamilbk**
This PR adds a number of improvements to the docs, namely Docker and SAML.

It also simplifies the number of ENV vars needed to be passed to the Docker container by moving the 
defaults into the app `config` files.

Fixes #913 
Fixes #960 
Fixes #959 
Fixes #794 
Fixes #746

### CODE REVIEW

File: docs/docs/administer/migrate.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+title: Migrate
+sidebar_position: 2
+---
+
+Chef Infra Client, the configuration system Chef Omnibus relies on, has been
+scheduled for End-of-Life in 2024. As such, Omnibus-based deployments
```
Comment (jamilbk) [inline docs/docs/administer/migrate.mdx]:
```suggestion
[scheduled for End-of-Life in 2024](https://docs.chef.io/versions/#supported-commercial-distributions).
As such, Omnibus-based deployments
```

File: docs/docs/administer/migrate.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+---
+title: Migrate
```
Comment (gongjason) [inline docs/docs/administer/migrate.mdx]: Should we name it "Migrate to Docker" so people don't mistake it for some other migration? Looks like we don't have an upgrade doc yet, but I can work on adding that.

**Review by gongjason — APPROVED** [review]

File: docs/docs/administer/migrate.mdx (hunk 2 above)
Comment (jamilbk) [inline docs/docs/administer/migrate.mdx]:
```suggestion
title: Migrate to Docker
```

Comment (jamilbk) [reply docs/docs/administer/migrate.mdx]: Upgrade for Docker should be pretty trivial -- can probably add it to the existing upgrade doc. Maybe via the `<Tabs>` thing if there's differences w/ Omnibus.

**Comment by jamilbk** [comment]
Merging to stage 0.6.0

**Pull request merged**